lib/kubernetes-deploy/cluster_resource_discovery.rb

```diff
@@ -2,16 +2,16 @@
 
 module KubernetesDeploy
   class ClusterResourceDiscovery
-    def initialize(namespace:, context:, logger:, namespace_tags:)
-      @namespace = namespace
-      @context = context
-      @logger = logger
+    delegate :namespace, :context, :logger, to: :@task_config
+
+    def initialize(task_config:, namespace_tags: [])
```

Comment:
Took a pass at using `task_config` this can be reverted if too much is going on in the pr

Reply:
What do you think about doing this the other way around: moving all of the stuff in this class into TaskConfig? Everything in here is effectively collected global context about the task's execution environment.

Reply:
I think this is a reasonable change for a follow-up PR. My original concern was that TaskConfig should actually be the holder of static information, but I don't actually think there would be issues with it having dynamic abilities.